Also check deviations for plot visibility & drawLine refactoring

egeerardyn opened the pull request:
When a line has deviations, it is always visible.
This should fix #812.

okomarov commented:
Can we add/edit an ACID test to have something like:

    errorbar(1:3,[1,1,1],'LineStyle','none')

egeerardyn commented:
@okomarov: perhaps not a bad idea.

miscco commented on src/matlab2tikz.m:
```diff
@@ -1613,18 +1613,15 @@ function assertRegularAxes(handle)
-    if ~isVisible(h)
-        return
-    end
-
```
Generally i am rather fond of exiting early. I am sure, that it has no runtime effect at all, but is there any particular reason you put it in the end?

egeerardyn replied:
It's hardly put in the end. Personally, I rather have 1 "if" per concept instead of first checking visibility and afterwards do a more thorough check on visibility (by means of markers and so on).

egeerardyn commented:
I cleaned up the code a bit, since the `drawLine` function looked ugly and did a few too many things.
To address the core of @miscco 's comment (i.e. return early), I extracted that check into another function, so now it reads quite a bit cleaner IMHO.

miscco commented:
:+1: Very nice
